File: algo_practice/leetcode/2000_reverse_prefix_of_word.ts
function reversePrefix(word: string, ch: string): string {
  const index = word.indexOf(ch);
  if (index === -1) return word;
  const prefix = word
    .substring(0, index + 1)
    .split("")
    .reverse()
    .join("");
  return prefix + word.substring(index + 1);
}

File: algo_practice/leetcode/2000_reverse_prefix_of_word.rb
# @param {String} word
# @param {Character} ch
# @return {String}
def reverse_prefix(word, ch)
  if word.include?(ch)
   index = word.index(ch)
   return word[0..index].reverse + word[index+1..word.length-1]
 else
   return word
  end
end

p reverse_prefix("abcdef", "d") # "dcbaef"
p reverse_prefix("xyxzxe", "z") # "zxyxxe"
p reverse_prefix("abcd", "z") # "abcd"
